### DISCUSSION TITLE
enum_as not working as expected

### DISCUSSION THREAD

**Discussion by mohantaylor** [SeaQL/sea-query#274] [Q&A] [answered]
## Description

Trying to get a count of a number of rows with an alias for the column.  Below is the example code that I was expecting to generate a select using the alias 'mycount' but the generated sql does not include this.

## Steps to Reproduce

```
let query = Query::select()
            .expr(
                Expr::expr(Expr::tbl(Data::Table, Data::Id).count()).as_enum(Alias::new("mycount")),
            )
            .from(Data::Table)
            .build(MysqlQueryBuilder);
        println!("{:?}", query);
```

### Expected Behavior

SQL output expected was:
```
SELECT COUNT(`data`.`id`) as mycount FROM `data`
```

### Actual Behavior

```
SELECT COUNT(`data`.`id`) FROM `data`
```

### Reproduces How Often

Always reproducible 

## Versions

sea-query = "0.23.0"

## Additional Information

**Comment by tyt2y3** [comment, marked as answer]
As enum is not intended for this, use https://docs.rs/sea-query/latest/sea_query/query/struct.SelectStatement.html#method.expr_as instead
